### PR TITLE
build: Replace GCC ARM Embedded with GNU Arm Embedded

### PR DESCRIPTION
**Pull request by carlescufi**
The old GCC ARM Embedded website on launchpad
(https://launchpad.net/gcc-arm-embedded) has been superseeded by the new
GNU Arm Embedded one
(https://developer.arm.com/open-source/gnu-toolchain/gnu-rm).

This also means a change of name from "GCC" to "GNU". Reflect this in
the enviroment variables so that the proper term is used henceforth.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
recheck

**Comment by carlescufi** [comment]
@dbkinder @nashif CI fails due to the amount of boards I changed, times out since it would take a long time to run all the tests on them.

File: doc/getting_started/getting_started.rst [hunk 1]
```diff
@@ -201,11 +201,11 @@ SDK, follow the steps below.
    .. code-block:: console
 
       # On Linux/macOS
-      export GCCARMEMB_TOOLCHAIN_PATH="~/gcc-arm-none-eabi-5_3-2016q1/"
-      export ZEPHYR_TOOLCHAIN_VARIANT=gccarmemb
+      export GNUARMEMB_TOOLCHAIN_PATH="~/gcc-arm-none-eabi-5_3-2016q1/"
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: Shouldn't all the references to filenames with ``gcc`` in them be changed to ``gnu`` too?

File: doc/getting_started/getting_started.rst [hunk 2]
```diff
@@ -201,11 +201,11 @@ SDK, follow the steps below.
...
+      export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
       # On Windows
-      set GCCARMEMB_TOOLCHAIN_PATH="%userprofile%\gcc-arm-none-eabi-5_3-2016q1\"
-      set ZEPHYR_TOOLCHAIN_VARIANT=gccarmemb
+      set GNUARMEMB_TOOLCHAIN_PATH="%userprofile%\gcc-arm-none-eabi-5_3-2016q1\"
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: another ``gcc`` reference to change?

File: doc/getting_started/getting_started.rst [hunk 3]
```diff
@@ -191,8 +191,8 @@ Using 3rd Party Cross Compilers
 To use a 3rd party cross compiler that is not provided by the Zephyr
 SDK, follow the steps below.
 
-#. We will use the `GCC ARM Embedded`_ compiler for this example, download the
-   package suitable for your operating system from the `GCC ARM Embedded`_ website
+#. We will use the `GNU ARM Embedded`_ compiler for this example, download the
+   package suitable for your operating system from the `GNU ARM Embedded`_ website
    and extract it on your file system. This example assumes the compiler was
    extracted to: :file:`<user folder>/gcc-arm-none-eabi-5_3-2016q1/`.
```
Comment (dbkinder) [inline doc/getting_started/getting_started.rst]: another ``gcc`` reference to change?

**Comment by carlescufi** [comment]
@SebastianBoe mind taking a look at this?

**Review by SebastianBoe — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
LGTM. Please send a heads up to the mailing list.

**Comment by carlescufi** [comment]
Merging with CI failing due to the amount of boards this change affects, which makes sanitycheck timeout

**Comment by carlescufi** [comment]
@MaureenHelm will do thanks!

**Pull request merged**